### DISCUSSION TITLE
Do something if a particular label is set

### DISCUSSION THREAD

**Discussion by pancetta** [community/community#26261] [Actions]
<p>Hi!</p>
<p>I just wrote my first GitHub Action, which gets triggered when an issue is opened. I now want to change this so that the runner starts only when a particular label is assigned. I know I can use the trigger/type “labeled” for issues, but I want to look for a specific label. Is that possible?</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Currently, GitHub Actions does <strong>NOT</strong> support to set <strong>labels filter</strong> for <strong>issues event</strong>.</p>
<p>However, as a workaround, you can use <strong>if conditional</strong> to skip all jobs in the workflow when the label does not match the specific label. And you need to add the <strong>if conditional</strong> on every job.</p>
<pre><code>jobs:
 job01:
    if: github.event.label.name == '&lt;label_name&gt;'
</code></pre>
<p>More details, you can reference here: <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fworkflow-syntax-for-github-actions%23jobsjob_idif&amp;data=02%7C01%7Cv-brira%40microsoft.com%7C886a6ceb19c4412da8be08d77d3e8fe8%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637115579963662357&amp;sdata=STF13AcDKUyEUbUKSJI%2FeveMSbI9RjTpjljg08UIzcI%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif</a></p>

**Comment by pancetta** [comment]
<p>Ah, I see, thanks a lot. I played a little with the <strong>if conditional</strong> and this works pretty well. It looks like that the job produces a “fail” if the condition is not met, right? This would give me an error message/email… I could add the conditional to each step and get rid of that, but this seems less obvious. Is the described behavior expected and intended?</p>
<p>Btw. I found another way to test for any given label (since I now knew where to look thanks to your reply):</p>
<pre><code>jobs:
  job01:
     if: contains( toJson(github), '&lt;label_name&gt;' )
</code></pre>
<p>Don’t know if that is a good idea (since it could find &lt;label_name&gt; somewhere else in the issue data), but so far it works.</p>

**Comment by BrightRan** [comment]
<p>Yeah, in the past, if all jobs in a workflow were skipped with the <strong>if conditional</strong> , the workflow displayed as “skipped/no status” status with a grey icon. But now, it displays as failed and will send the notification email for the failed workflow run.</p>

**Comment by lmancilla** [comment]
<p>Or you can use:</p>
<pre><code>if: contains(github.event.pull_request.labels.*.name, 'some-defined-label')
</code></pre>
<p>It works in the case you put a label in a PR.</p>

**Reply by qrkourier** [reply]
I'm trying to find a way to evaluate a PR label in the push event that's triggered when the PR is merged.

**Reply by jbcallej** [reply]
Have you been able to find a way to do that?

**Comment by Djiit** [comment]
<p>This is a great solution ! We are using this but now if shows a failed workflow instead of just not running it. Do you have a workaround ?</p>

**Comment by Et7f3** [comment]
<p>add a empty step that just echo some dummy text.</p>
<p>Edit: Sorry I haven’t seen the date.</p>

**Comment by Maciul** [comment]
<p>Hello!</p>
<p>I have been trying to get labels in Github Actions with all the methods above and unable to do so. Did something change in the way those can be extracted from the PR?</p>
<p>Currently, my actions trigger on push and labels are missing in the Github context <code>toJson(github)</code>. The only label property is <code>labels_url</code>.</p>
<p>Any help would be appreciated!</p>
<p>Cheers</p>

**Reply by qrkourier** [reply]
I am trying to solve this same problem. I do not see a `labels_url` property on the push event that is triggered when the PR containing the labels that I wish to evaluate is merged.

**Comment by y0urself** [comment]
<p>Hey. I searched for a while, but did not found any solution.</p>
<p>Is it possible to check if the label was set by some specific account or even better by the codeowner(s)?</p>

**Comment by joshuat** [comment]
<p>This is really disappointing - not being able to limit a workflow from running ends up polluting the list of ran actions pretty dramatically, because even if you conditionalize a job it will still show up in the list of actions just with a gray cross over it indicating it didn’t run.</p>
<p>This is exceptionally problematic when conditionally running a large number of actions.</p>

**Comment by WarWolf89** [comment]
Just to save someone else some time:
Scenario is that I only want to trigger a lint job if the file changed is of a specific extension(our case is golang). The workflow should only trigger if the PR is approved.

### Problem 1 
As described [here](https://stackoverflow.com/questions/70438914/trigger-a-github-action-on-pull-request-approval-and-path) 
> The ON trigger conditions work as OR and not as AND. Therefore, in the question sample, the workflow will trigger when a pull_request_review is submitted OR when one of the updated file(s) path is the one informed.

So using path in 
```
on:
  pull_request_review:
    branches:
      - main
    types: [submitted]
```
is out of the question unfortunately. BTW the code over at [this](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request) example won't work either for all file types changed, the correct glob syntax will be `**.*.go`, but the path keyword will be evaluated.

We're getting around this by using the [labeler](https://github.com/actions/labeler) action.

The lint workflow now looks something like:

```
.....
jobs:
  lint:
    # the ON keyword is ANY not AND for conditions regarding path. Need to work around with labelling for PRs. 
    if: contains(github.event.pull_request.labels.*.name, 'golang') && github.event.review.state == 'approved'
.....
```

### Problem 2
There's a major difference between how `pull_request` and `pull_request_review` events are defined. 
`github.event.label.name == 'project'` will work for event type `pull_request` **BUT** the real unexpected case happens when you open PR's that for example change files with the .md extension: 

 > **the workflow will also trigger for those if at any point in time there was a PR with the golang label opened on the repo!!!!** 

I bashed my head against this for about an hour trying to figure out how it's evaluating to true, and with some trial and error found that with @lmancilla 's original suggestion of `if: contains(github.event.pull_request.labels.*.name, 'some-defined-label')` works as expected. 
 
 ### TLDR 
Be mindful of which exact `pull_request` event type you're referencing because the event object structure is different. Use `github.event.label.name == 'project'` for `pull_request` triggers use `contains(github.event.pull_request.labels.*.name, 'some-defined-label')` for `pull_request_review`.
